test/unit/app/users/postEditServices.test.js

```diff
@@ -132,5 +139,88 @@ describe("when hitting the post function of edit service", () => {
         roles: ["There has been an error"],
       },
     });
+    expect(res.redirect).toHaveBeenCalledTimes(0);
+  });
+
+  it("should display an error message if saving the session failed", async () => {
+    ((req.session.save = jest.fn((cb) => cb("Something went wrong"))),
+      await postEditService(req, res));
+
+    expect(res.render.mock.calls[0][1]).toMatchObject({
+      validationMessages: {
+        roles: "Something went wrong saving session data, please try again",
+      },
+    });
+    expect(logger.error.mock.calls).toHaveLength(1);
+    expect(logger.error.mock.calls[0][0]).toBe(
+      "An error occurred when saving to the session",
+    );
+  });
+
+  it("should give an empty array to policy engine if no roles are selected", async () => {
+    req.body.role = undefined;
+
+    await postEditService(req, res);
+
+    expect(policyEngine.validate.mock.calls[0][3]).toStrictEqual([]);
+  });
+
+  it("should give an array of one role to policy engine if one role selected", async () => {
+    req.body.role = "role1";
+
+    await postEditService(req, res);
+
+    expect(policyEngine.validate.mock.calls[0][3]).toStrictEqual(["role1"]);
+  });
+
+  it("should give an array of multiple roles to policy engine if multiple roles selected", async () => {
+    req.body.role = ["role1", "role2"];
+
+    await postEditService(req, res);
+
+    expect(policyEngine.validate.mock.calls[0][3]).toStrictEqual([
+      "role1",
+      "role2",
+    ]);
+  });
+
+  it("should redirect if there are no errors returned from policy engine", async () => {
+    policyEngine.validate.mockReturnValue([]);
+
+    await postEditService(req, res);
+
+    expect(res.render.mock.calls).toHaveLength(0);
+    expect(res.sessionRedirect).toHaveBeenCalledTimes(1);
+    expect(res.sessionRedirect.mock.calls[0][0]).toBe(
+      "service1/confirm-edit-service",
+    );
+  });
+
+  it("should add manage_users to the url is 'isUserMangement' returns true", async () => {
+    isUserManagement.mockReset().mockReturnValue(true);
+    policyEngine.validate.mockReturnValue([]);
+
+    await postEditService(req, res);
+
+    expect(res.render.mock.calls).toHaveLength(0);
+    expect(res.sessionRedirect).toHaveBeenCalledTimes(1);
+    expect(res.sessionRedirect.mock.calls[0][0]).toBe(
+      "service1/confirm-edit-service?manage_users=true",
+    );
+  });
+
+  it("should add manage_users to the url is 'isUserMangement' returns true", async () => {
```

Comment:
Same name as above test. Consider renaming to "should redirect to subService requests when RID and REVIEW_SUBSERVICE_REQUEST are set"